src/Nethermind/Nethermind.Abi/AbiArray.cs

```diff
@@ -36,14 +37,24 @@ public override (object, int) Decode(byte[] data, int position, bool packed)
 
         public override byte[] Encode(object? arg, bool packed)
         {
-            if (arg is Array input)
+            int length;
+            byte[][] encodedItems;
+            switch (arg)
             {
-                byte[][] encodedItems = EncodeSequence(input.Length, ElementTypes, input.Cast<object?>(), packed, 1);
-                encodedItems[0] = UInt256.Encode((BigInteger)input.Length, packed);
-                return Bytes.Concat(encodedItems);
+                case Array array:
```

Comment:
Why those switch case here? Isn't `IList` alone enough? If it is, I'd suggest to go [this way](https://github.com/NethermindEth/nethermind/compare/feature/aura-withdrawals...feature/aura-withdrawals-array-pool-enumerable?expand=1#diff-fad6acb1b3a4e17d84d6444e7665017e720ef51c84659a38437f15763daa945b).

Reply:
IEnumerable is 2x enumeration and allocation for enumerators (minor), IList is theoretically fine as Array supports it, just a tad slower.

Reply:
So do you like to keep this as it is?

Reply:
Nevermind. Benchmarking showed that using `Count()` for `IEnumerable<>` degrades performance more than 10 times and allocates 2 times more! Otherwise, a negligible difference.